features/FEATURE_LWIP/lwip-interface/ppp_lwip.cpp

```diff
@@ -390,7 +398,19 @@ nsapi_error_t nsapi_ppp_connect(FileHandle *stream, Callback<void(nsapi_event_t,
 
 nsapi_error_t nsapi_ppp_disconnect(FileHandle *stream)
 {
-    return my_interface->bringdown();
+    if (my_stream != stream) {
+        return NSAPI_ERROR_NO_CONNECTION;
+    }
+
+    nsapi_error_t retcode = my_interface->bringdown();
+
+    connection_status_cb = NULL;
+    /* Detach callbacks, and put handle back to default blocking mode */
+    my_stream->sigio(NULL);
+    my_stream->set_blocking(true);
+    my_stream = NULL;
+
+    return retcode;
```

Comment:
AT_CellularNetwork::disconnect should probably take ownership also in case of if (err != NSAPI_ERROR_OK)

Reply:
true, fixed to AT_Cellularnetwork::disconnect()